xfrm_policy.go

```diff
@@ -52,6 +52,9 @@ type XfrmPolicyTmpl struct {
 type XfrmPolicy struct {
 	Dst      *net.IPNet
 	Src      *net.IPNet
+	Proto    Proto
+	DPort    uint16
```

Comment:
All of the other port constructs use int instead of u1nt16 for the higher level representation, so these should be changed to match. I'm also wondering if they should  be called DstPort and SrcPort to match State.